File: moforum/src/main/java/com/moforum/entity/Post.java
package com.moforum.entity;

import lombok.Data;

import java.time.LocalDateTime;

/**
 * 作者:Momo同学
 * 日期: 2026/4/12 18:43
 */
@Data
public class Post {
    private Long id;
    private Long userId;
    private Long boardId;
    private String title;
    private String content;
    private Integer viewCount;
    private Integer replyCount;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
    /** 联表查询：楼主昵称 */
    private String authorName;
    /** 联表查询：楼主头像 */
    private String avatarUrl;
    /** 联表查询：吧名 */
    private String boardName;
}

File: moforum-web/src/api/post.ts
import { http } from './http'

export type PostRow = {
  id: number
  userId: number
  boardId?: number
  title: string
  content: string
  viewCount?: number
  replyCount?: number
  createTime?: string
  updateTime?: string
  authorName?: string
  avatarUrl?: string
  boardName?: string
}

export type PostPage = {
  list: PostRow[]
  total: number
  page: number
  pageSize: number
}

export type CreatePostRes = { success: boolean; message: string }

export type DeletePostRes = { success: boolean; message: string }

export function createPost(payload: {
  boardId: number
  title: string
  content: string
}) {
  return http.post<CreatePostRes>('/post/create', payload)
}

export function listPostsByBoard(boardId: number, page: number, pageSize: number) {
  return http.get<PostPage>('/post/list', { params: { boardId, page, pageSize } })
}

export function listPostsByUser(userId: number, page: number, pageSize: number) {
  return http.get<PostPage>('/post/list', { params: { userId, page, pageSize } })
}

export function latestPosts(limit = 8) {
  return http.get<PostRow[]>('/post/latest', { params: { limit } })
}

export function getPostDetail(id: number) {
  return http.get<PostRow | null>('/post/detail', { params: { id } })
}

export function deletePost(postId: number) {
  return http.post<DeletePostRes>('/post/delete', { postId })
}
